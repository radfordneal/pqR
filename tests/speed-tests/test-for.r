source("time.r")
 
test.name <- "for"

# Timing tests:

f <- test.cmp (function (n) 
  { a <- 0; for (j in 1:n) for (i in 1:n) a <- a+i/j; a })

sys.time(r <- f(2000))
print(r)

f <- test.cmp (function (n) 
  { a <- 0; for (k in 1:n) for (j in 1:n) for (i in 1:n) a <- a+k*i/j; a })

sys.time(r <- f(150))
print(r)

f <- test.cmp (function (n) 
  { a <- 0; 
    for (l in 1:n) for (k in 1:n) for (j in 1:n) for (i in 1:n) a <- a+l*k*i/j
    a
  })

sys.time(r <- f(40))
print(r)

g <- test.cmp (function (n) 
  { a <- 0; for (j in seq_len(n)) for (i in seq_len(n)) a <- a+i/j; a })

sys.time(r <- g(2000))
print(r)

f <- test.cmp (function (n) 
  { for (l in 1:n) for (k in 1:n) for (j in 1:n) for (i in 1:n) i })

sys.time(r <- f(70))
print(r)

v1n <- 1:70

f <- test.cmp (function () 
  { for (l in v1n) for (k in v1n) for (j in v1n) for (i in v1n) i })

sys.time(r <- f())
print(r)

f <- test.cmp (function () 
{ v1 <- 1:70; 
  for (l in v1) 
  { v2 <- 1:70
    for (k in v2) 
    { v3 <- 1:70
      for (j in v3) 
      { v4 <- 1:70
        for (i in v4) i 
      }
    }
  }
})

sys.time(r <- f())
print(r)


# A quick correctness check:

for (i in 1:2) { print(i); if (i==1) a <- i; print(a); }
cat("\n")
