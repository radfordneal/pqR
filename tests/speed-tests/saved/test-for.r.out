
---- for 

Time:
[1] 16364915
Time:
[1] 717100361
Time:
[1] 2359051718
Time:
[1] 16364915
Time:
NULL
Time:
NULL
Time:
NULL
[1] 1
[1] 1
[1] 2
[1] 1

